Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
hehehe no problem, that's the issue, is not about those functions it's about to who those functions will be applied to, let's say we add the logic of `ArrayPartiallyDecodable` to any array of Bools, we could do that right? the problem will be that you will not have any way to distinguish if the user wants that for just a particular array of Bools or.. for ANY array of Bools since once you apply `ArrayPartiallyDecodable` to the primitive type then it will apply for **Any**